Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgePaymentMethodsViewController.swift

```diff
@@ -172,11 +172,16 @@ final class PledgePaymentMethodsViewController: UIViewController {
           strongSelf.messageDisplayingDelegate?
             .pledgeViewController(strongSelf, didErrorWith: error.localizedDescription)
         case let .success(paymentSheetFlowController):
-          strongSelf.paymentSheetFlowController = paymentSheetFlowController
-          strongSelf.paymentSheetFlowController?.presentPaymentOptions(from: strongSelf) { [weak self] in
-            guard let strongSelf = self else { return }
+          let topViewController = strongSelf.navigationController?.topViewController
+          let paymentSheetShownWithinPledgeContext = topViewController is PledgeViewController
 
-            strongSelf.confirmPaymentResult(with: data.clientSecret)
+          if paymentSheetShownWithinPledgeContext {
```

Comment:
Tested and this is working as expected. I'm confused why this logic was being hit though.

I would think that `PaymentSheet.FlowController.create` wouldn't finish executing if the backer hits the Back button before the sheet is finished creating/loading. 
I've set some breakpoints in the `PaymentSheet.FlowController.create` flow as well as around your changes in this completion handler and can't seem to hit them when backing out prematurely.